### PR TITLE
Feature - Edit Employee

### PR DESCRIPTION
**Pull request by at-hangtran**

### CODE REVIEW

File: src/test/java/stepdefs/employee/EmployeesDefinitions.java [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package stepdefs.employee;
 
+import cucumber.api.PendingException;
```
Comment (at-vinhhuynh) [inline src/test/java/stepdefs/employee/EmployeesDefinitions.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/stepdefs/employee/EmployeesDefinitions.java#L3"></span>Unused import - cucumber.api.PendingException.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/object/Employee.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ */
+public class Employee {
+    private String name;
+    private String code;
+    private String email;
+    private String manager;
+    private String team;
+    private String group;
+
+    public Employee(String name, String code, String email, String manager, String team, String group) {
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/object/Employee.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/object/Employee.java#L14"></span>Parameter name should be final.</td>
    </tr>
  </tbody>
</table>

Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/object/Employee.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/object/Employee.java#L14"></span>Parameter code should be final.</td>
    </tr>
  </tbody>
</table>

Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/object/Employee.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/object/Employee.java#L14"></span>Parameter email should be final.</td>
    </tr>
  </tbody>
</table>

Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/object/Employee.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/object/Employee.java#L14"></span>Parameter manager should be final.</td>
    </tr>
  </tbody>
</table>

Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/object/Employee.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/object/Employee.java#L14"></span>Parameter team should be final.</td>
    </tr>
  </tbody>
</table>

Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/object/Employee.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/object/Employee.java#L14"></span>Parameter group should be final.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/object/Employee.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        this.name = name;
+        this.code = code;
+        this.email = email;
+        this.manager = manager;
+        this.team = team;
+        this.group = group;
+    }
+
+    public String getName() {
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/object/Employee.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/object/Employee.java#L23"></span>Method ‘getName’ is not designed for extension - needs to be abstract, final or empty.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/object/Employee.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        return name;
+    }
+
+    public String getCode() {
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/object/Employee.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/object/Employee.java#L27"></span>Method ‘getCode’ is not designed for extension - needs to be abstract, final or empty.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/object/Employee.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        return code;
+    }
+
+    public String getEmail() {
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/object/Employee.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/object/Employee.java#L31"></span>Method ‘getEmail’ is not designed for extension - needs to be abstract, final or empty.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/object/Employee.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        return email;
+    }
+
+    public String getManager() {
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/object/Employee.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/object/Employee.java#L35"></span>Method ‘getManager’ is not designed for extension - needs to be abstract, final or empty.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/object/Employee.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        return manager;
+    }
+
+    public String getTeam() {
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/object/Employee.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/object/Employee.java#L39"></span>Method ‘getTeam’ is not designed for extension - needs to be abstract, final or empty.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/object/Employee.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        return team;
+    }
+
+    public String getGroup() {
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/object/Employee.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/object/Employee.java#L43"></span>Method ‘getGroup’ is not designed for extension - needs to be abstract, final or empty.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/page/employee/EmployeesPage.java [hunk 9]
```diff
@@ -391,4 +421,42 @@ private boolean isElementFound(final WebElement cell, final int column) {
             return false;
         }
     }
+
+    private String getMoreTeamOrGroup(boolean isTeam) {
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/page/employee/EmployeesPage.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/page/employee/EmployeesPage.java#L425"></span>Parameter isTeam should be final.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java [hunk 10]
```diff
@@ -0,0 +1,360 @@
+package vn.asiantech.page.employee.editemployee;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openqa.selenium.*;
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java#L4"></span>Using the ‘.<em>’ form of import should be avoided - org.openqa.selenium.</em>.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java [hunk 11]
```diff
@@ -0,0 +1,360 @@
...
+    @CacheLookup
+    private WebElement infoContactContainer;
+
+    private WebDriver driver;
+
+    public EditEmployeePage(final WebDriver driver) {
+        this.driver = driver;
+    }
+
+    @Override
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java#L54"></span>Method ‘navigateTo’ is not designed for extension - needs to be abstract, final or empty.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java [hunk 12]
```diff
@@ -0,0 +1,360 @@
...
+    public EditEmployeePage navigateTo(WebDriver webDriver) {
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java#L55"></span>Parameter webDriver should be final.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java [hunk 13]
```diff
@@ -0,0 +1,360 @@
...
+    }
+
+    public final void chooseDateToDeactivate() {
+        WebElement dateTime = driver.findElement(By.cssSelector("p-calendar[name=deactivate_date]"));
+        dateTime.click();
+        dateTime.findElement(By.xpath("//td[not(contains(@class,'ui-datepicker-other-month'))]")).click();
+    }
+
+    public final void clickDeactiveButtonOfDialog() {
+        WebElement btnDeactive = dialog.findElement(By.cssSelector(".btn-submit"));
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java#L162"></span>‘btnDeactive’ hides a field.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java [hunk 14]
```diff
@@ -0,0 +1,360 @@
...
+    public final void clickNewEventButton() {
+        WebElement btnNewEvent = driver.findElement(By.cssSelector(".btn.btn-default.btn-add.btn-block.ng-star-inserted"));
+        waitForElement(driver, btnNewEvent);
+        btnNewEvent.click();
+    }
+
+    public final void chooseTypeOfEvent(final String event) {
+        dialog.findElement(By.xpath("//div[contains(@class,'ui-dropdown ui-widget ui-state-default ui-corner-all ui-helper-clearfix')]")).click();
+        int position = 0;
+        switch (event) {
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java#L195"></span>switch without “default” clause.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java [hunk 15]
```diff
@@ -0,0 +1,360 @@
...
+
+    public final void clickDeleteNewEvent() {
+        List<WebElement> events = timeLineContainer.findElement(By.cssSelector(".timeline-item.m-t-md")).findElements(By.cssSelector(".row.ng-star-inserted"));
+        if (events.size() > 1) {
+            events.get(0).findElement(By.className("btn-delete")).click();
+        }
+    }
+
+    public final boolean isConfirmDialogDisplayed() {
+        WebElement dialog = driver.findElement(By.id("static-dialog-wrapper"));
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java#L262"></span>‘dialog’ hides a field.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java [hunk 16]
```diff
@@ -0,0 +1,360 @@
...
+        waitForElement(driver, dialog);
+        return dialog.isDisplayed();
+    }
+
+    public final void clickSubmitButton() {
+        WebElement dialog = driver.findElement(By.id("static-dialog-wrapper"));
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java#L268"></span>‘dialog’ hides a field.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java [hunk 17]
```diff
@@ -0,0 +1,360 @@
...
+    public final EditEmployeePage fillInfoContact(final String info) {
+        fillDataRelative(By.cssSelector("input[formcontrolname=title]"), info);
+        return this;
+    }
+
+    public final void fillName(final String name) {
+        fillDataRelative(By.cssSelector("input[formcontrolname=name]"), name);
+    }
+
+    public final void fillPhone(String phone) {
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java#L285"></span>Parameter phone should be final.</td>
    </tr>
  </tbody>
</table>

File: src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java [hunk 18]
```diff
@@ -0,0 +1,360 @@
...
+        if (positions.size() > 1) {
+            positions.get(new Random().nextInt(positions.size() - 1) + 1).click();
+        }
+    }
+
+    private void chooseDate() {
+        WebElement dayTimeContainer = dialog.findElement(By.cssSelector(".ui-calendar"));
+        dayTimeContainer.click();
+        new WebDriverWait(driver, Constant.DEFAULT_TIME_OUT).until(
+                driver -> dayTimeContainer.findElement(By.xpath("..")).getAttribute("class").contains("ui-inputwrapper-focus"));
```
Comment (at-vinhhuynh) [inline src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/selenium-test/blob/513d50aa5a3067d808d34920dc0267c9612605af/src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java#L343"></span>‘driver’ hides a field.</td>
    </tr>
  </tbody>
</table>

**Comment by at-vinhhuynh** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR, try to keep changes smaller if you can</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java [hunk 19]
```diff
@@ -0,0 +1,362 @@
...
+    public EditEmployeePage(final WebDriver driver) {
+        this.driver = driver;
+    }
+
+    @Override
+    public final EditEmployeePage navigateTo(final WebDriver webDriver) {
+        return null;
+    }
+
+    public final boolean isDeactiveButtonClickable() {
```
Comment (at-vietphan) [inline src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java]: replace isDeactiveButtonClickable to isDeactivateButtonClickable and more func have name Deactive

File: src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java [hunk 20]
```diff
@@ -0,0 +1,362 @@
...
+
+    public final boolean isPhoneErrorMessageDisplayed() {
+        List<WebElement> relatives = infoContactContainer.findElements(By.xpath("//tr[contains(@class,'ng-dirty') or contains(@class,'ng-pristine')]"));
+        if (relatives.size() > 1) {
+            return relatives.get(relatives.size() - 1).findElement(By.cssSelector("input[formcontrolname=phone]")).getAttribute("class").contains("ng-invalid");
+        }
+        return false;
+    }
+
+    private boolean isButtonDisplayed(final WebElement button) {
```
Comment (at-vietphan) [inline src/test/java/vn/asiantech/page/employee/editemployee/EditEmployeePage.java]: Please use func isElementPresented in BasePage instead func isButtonDisplayed

**Comment by at-vietphan** [comment]
LGTM 👍

**Pull request merged**